### PR TITLE
Graph adjacency matrix implementation

### PR DESCRIPTION
**Pull request by MohitPatni0731**

### CODE REVIEW

**Comment by MohitPatni0731** [comment]
Remove vertex and print graph function implementation is pending. Will do it by tommorow morning

File: Graph/main.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#include <cassert>
+using namespace std;
+
+class Graph {
+public:
+    int vertices;
+    vector<vector<int>> adjacency_matrix;
+
+public:
+    Graph(int vertices) : vertices(vertices), adjacency_matrix(vertices, vector<int>(vertices, 0)) {}
```
Comment (patniharshit) [inline Graph/main.cpp:13]: In CPP syntax is bit hard to remember. Its ok to look up how to do something the first time but next time try to remember how to do it.

In an interview it won't be possible to look up anything.

Comment (patniharshit) [reply Graph/main.cpp:13]: For example this is equivalent of init function in Python, it slightly more complicated, make sure to understand what it is doing instead of just copy pasting.

File: Graph/main.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    int size_of_vertices() {
+        return vertices;
+    }
+
```
Comment (patniharshit) [inline Graph/main.cpp:32]: I am not sure how these two function which are relying on vertices are working. 

I don't see vertices variable being incremented anywhere during insert.

Comment (patniharshit) [reply Graph/main.cpp:32]: Do we need to keep track of vertices?

File: Graph/main.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    cout << "1: test_add_edge() passed!" << endl; 
+}
+
+bool test_is_empty() {
+    Graph graph(2);
+    graph.add_edge(0, 1);
+    assert(graph.is_empty()==false);
+    cout << "2: test_is_empty() passed!" << endl;
+    return true;
+}
```
Comment (patniharshit) [inline Graph/main.cpp:75]: Missing test case to check empty graph.

File: Graph/main.cpp [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                }
+            }
+        }
+        return count;
+    }
+
+    void remove_edge(int node1, int node2) {
+        adjacency_matrix[node1][node2] = 0;
+        adjacency_matrix[node2][node1] = 0;
+    }
```
Comment (patniharshit) [inline Graph/main.cpp:48]: What happens when someone tries to remove an edge that doesn't exists?

**Review by patniharshit — COMMENTED** [review]
Missing time and space complexities in fucntions

File: Graph/main.cpp (hunk 1 above)
Comment (patniharshit) [reply Graph/main.cpp:13]: Here this is 
```
Constructor Initialization: Use of an initializer list in the constructor to initialize class members.
```
(read about this from our CPP book again)

Comment (patniharshit) [reply Graph/main.cpp:13]: Another thing here is nester vecotr/2d vector - remember this syntax from next time.
